tap/tcp_reader.go

```diff
@@ -81,14 +79,23 @@ func (reader *tcpReader) sendMsgIfNotClosed(msg api.TcpReaderDataMsg) {
 	reader.Unlock()
 }
 
+func (reader *tcpReader) isProtocolIdentified() bool {
+	return reader.parent.protocol != nil
+}
+
 func (reader *tcpReader) Read(p []byte) (int, error) {
+	if reader.exhaustBuffer {
+		reader.data = reader.buffer
```

Comment:
I'm worried that copying a slice could have unexpected results when appending. Because both slices are views to the same underlaying array, appending to one could overwrite appends to the other.
Not sure if this is the case here or if go is smart enough to avoid this effect.

Reply:
@nimrod-up9 fixed with https://github.com/up9inc/mizu/commit/11e535cd27eaff11895cba920d6d857739d7ad20